### PR TITLE
cleanup vdlp

### PR DESCRIPTION
**Pull request by trapexit**
Part of #38

### CODE REVIEW

File: libfreedo/vdlp.h [hunk 1]
```diff
@@ -26,28 +26,47 @@
...
-*/
+  */
 
 /* VDLP.h: interface for the CVDLP class. */
 
-#ifndef	VDLP_3DO_HEADER
-#define VDLP_3DO_HEADER
-
-#include "freedocore.h"
+#pragma once
```
Comment (twinaphex) [inline libfreedo/vdlp.h]: Can we try to avoid this and instead just do simple header guards?

File: libfreedo/hack_flags.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma once
```
Comment (twinaphex) [inline libfreedo/hack_flags.h]: I suggest we avoid using pragma once just for consistency reasons and instead just make plain header guards instead.

File: libfreedo/vdlp.h (hunk 1 above)
Comment (trapexit) [reply libfreedo/vdlp.h]: Sure

File: libfreedo/hack_flags.h (hunk 2 above)
Comment (trapexit) [reply libfreedo/hack_flags.h]: It's already in several other files. Let me merge this and then change them all in a separate commit. I'm assuming we shouldn't be modifying dep code which uses it? (libchdr)

**Comment by twinaphex** [comment]
OK, I will have a look at libchdr later.

**Comment by trapexit** [comment]
Shouldn't external libs be left alone generally? Most compilers support pragma once. Doesn't sound like this is a combatibility problem.

**Comment by orbea** [comment]
@trapexit If you have any libchdr changes can you please submit them upstream too?

https://github.com/rtissera/libchdr

Upstream has been accepting of all libretro changes I have submitted so far. :)

**Comment by twinaphex** [comment]
@trapexit I had to make several edits to libchdr in the first place to make them conducive for our libretro cores.

I keep track of all external dependencies here that have ever been used in a libretro core -

https://github.com/libretro/libretro-deps

Libretro cores have certain requirements when it comes to being able to have no real system library dependencies and whatnot. Just using dependencies unaltered is a no-go and I wouldn't want to be playing dll/solib hell anytime soon. So this is a deliberate design strategy to make sure that cores are as self-contained as possible without having to require external libs.

**Comment by trapexit** [comment]
Will do. It's really a matter of taste with regard to include guards. I tend to use pragma once because it's simpler and pretty widely supported but no need to change libchdr if they are already happy with what they are doing.

**Comment by twinaphex** [comment]
Yeah, let's leave it for what it is for now then as far as libchdr is concerned.

**Comment by trapexit** [comment]
I understand the need for limited dependencies. I was only speaking to the use of pragma once. If there is a need to work on the deps I'll be sure to attempt to upstream them.

**Comment by orbea** [comment]
@twinaphex fwiw, I backported several of your changes to libchdr and they were merged without issue. That said I still need to finish backporting most of the other changes, but at least last time I looked the diff between upstream and libretro wasn't that big.

**Pull request merged**